core/controllers/editor.py

```diff
@@ -638,13 +638,18 @@ def post(self, entity_type, entity_id):
 
         allowed_formats = ', '.join(
             list(feconf.ACCEPTED_IMAGE_FORMATS_AND_EXTENSIONS.keys()))
-
-        if filename is not None and filename.endswith('.svg'):
+        if html_validation_service.is_xml_parsable(raw):
             file_format = 'svg'
-            invalid_tags = html_validation_service.get_invalid_svg_tags(raw)
-            if len(invalid_tags) > 0:
+            invalid_tags, invalid_attrs = (
+                html_validation_service.get_invalid_svg_tags_and_attrs(raw))
+            if invalid_tags or invalid_attrs:
+                invalid_tags_message = (
+                    'tags: %s' % invalid_tags if invalid_tags else '')
+                invalid_attrs_message = (
```

Comment:
As before -- note that what counts as a valid/invalid attr should be tag-specific.